kyleteeter/github-gists-app
# Create Search Bar UI In React (forgot to work off branch so most commit are in master)

### Create Search Bar UI In React
The goal of this first Pull Request is to spin up a new create-react-app and initialize the Search Bar UI. 

The design mockup I was given was this:
![design-mockup](https://user-images.githubusercontent.com/29210839/146564026-56e93041-a472-44b9-81b4-23b109c93568.jpg)

The design that I have created is in this Pull Request is:
![App Design](https://user-images.githubusercontent.com/29210839/146564551-35df914a-99e0-4b96-bd5b-21f764269dc4.png)

The component breakdown for this app has been set up like this:
![Component Breakdown (1)](https://user-images.githubusercontent.com/29210839/146565281-0f968e8d-c683-43dd-b8f0-ca180f4b9062.png)

1. Wrapper (yellow): contains the entirety of the app and will store Gist results
2. SearchBar (light blue): collects username and will eventually retrieve Gists
3. SearchButtons (purple): reusable buttons
4. LogoSection (dark blue): contains the logo image and text
5. SearchResults (red): will eventually display all the search responses 

### Testing
1. Make sure that the desktop view of the app closely resembles the design. 
2. Verify that the app is functional and styled correctly for mobile users. 

### _Notes_
For most of the styled-components, I used 'em' for sizing instead of 'px'. However, I do have a few instances of using 'px' is it better to do everything exclusively with 'em'?
